### ISSUE TITLE
Return warnings instead of errors

### ISSUE THREAD

**Issue by ThomasKientz** [vuejs/eslint-plugin-vue#1004]
**Tell us about your environment**	
* **ESLint version: v5.16.0**	
* **eslint-plugin-vue version: 5.2.3 (same with 6+)**	
* **Node version: v12.1.0**	

**Please show your full configuration:**

```json
module.exports = {
  root: true,
  env: {
    node: true
  },
  extends: ["plugin:vue/essential", "@vue/prettier"],
  rules: {
    "no-console": process.env.NODE_ENV === "production" ? "error" : "off",
    "no-debugger": process.env.NODE_ENV === "production" ? "error" : "off"
  },
  parserOptions: {
    parser: "babel-eslint"
  }
};

```

**What did you do?**	

```vue
<template>
  <div id="app">
    <img alt="Vue logo" src="./assets/logo.png" />
    <!-- <HelloWorld msg="Welcome to Your Vue.js App" /> -->
  </div>
</template>

<script>
import HelloWorld from "./components/HelloWorld.vue";

export default {
  name: "app",
  components: {
    HelloWorld
  }
};
</script>
```

**What did you expect to happen?**
I expected an eslint **warning**

**What actually happened?**

> ```
> error: The "HelloWorld" component has been registered but not used (vue/no-unused-components) at src/App.vue:14:5:
>   12 |   name: "app",
>   13 |   components: {
> > 14 |     HelloWorld
>      |     ^
>   15 |   }
>   16 | };
>   17 | </script>
> ```
> 

[Minimum reproductible project ](https://github.com/ThomasKientz/vue-eslint-issue-repro)

**Comment by ota-meshi** [comment]
```json
"vue/no-unused-components" : "warning"
```

**Comment by ota-meshi** [comment]
Thank you for this issue.

You can set and change rules as follows:

```json
"rules": {
  "vue/no-unused-components" : "warn",
}
```
